patientsearch/api.py

```diff
@@ -241,15 +244,38 @@ def resource_bundle(resource_type):
 
     """
     token = validate_auth()
+    # Check for the store's configurations
+    active_patient_flag = current_app.config.get("ACTIVE_PATIENT_FLAG")
+    params = dict(deepcopy(request.args))  # Necessary on ImmutableMultiDict
+
+    # Override if the search is specifically for inactive objects, only occurs
+    # when working on reactivating a patient
+    inactive_search_param = request.args.get("inactive_search")
+    is_inactive_search = False
+    if inactive_search_param:
+        is_inactive_search = inactive_search_param.lower() in {"true", "1"}
+    if is_inactive_search:
+        active_patient_flag = False
+    # need to remove this param, name/key error is raised when passed to HAPI_request
+    if inactive_search_param is not None:
+        del params["inactive_search"]
```

Comment:
quite a bit of logic here to determine if `params["active"]` should be set to "true".  if this can't be simplified, let's put it in a local function.

ideally you could use the same function for similar logic in several APIs, which would also make it easier to test.  if it can't be generalized, you can nest the function.